src/main/java/core/basesyntax/service/impl/FruitServiceImpl.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitService;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class FruitServiceImpl implements FruitService {
+    private static final int OPERATION_INDEX = 0;
+    private static final int FRUIT_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+    private final OperationStrategy operationStrategy;
+
+    public FruitServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void processDailyActivities(List<String> data) {
+        for (String activity : data) {
+            String operation = activity.split(",")[OPERATION_INDEX];
+            String fruit = activity.split(",")[FRUIT_INDEX];
+            String quantity = activity.split(",")[QUANTITY_INDEX];
+
+            FruitTransaction transaction = new FruitTransaction(operation, fruit, quantity);
+
+            OperationHandler operationHandler = operationStrategy.getOperationHandler(transaction);
+            operationHandler.registerTransaction(transaction);
```

Comment:
parsing FruitTransaction from string and FruitTransaction executing are different operations. Do not mix them in one method. Create separate TransactionParser with `List<FruitTransaction> parse(List<String> data)` method. Call it in main().